examples/thaumcraft.groovy

```diff
@@ -0,0 +1,110 @@
+import net.minecraft.item.ItemStack
+import net.minecraft.init.Blocks
+import net.minecraft.util.ResourceLocation
+import thaumcraft.api.aspects.Aspect
+import thaumcraft.api.aspects.AspectList
+import thaumcraft.api.ThaumcraftApiHelper
+import com.cleanroommc.groovyscript.compat.mods.thaumcraft.AspectWrapper
+import com.cleanroommc.groovyscript.compat.mods.thaumcraft.Warp
+import com.cleanroommc.groovyscript.compat.mods.thaumcraft.LootBag
+import com.cleanroommc.groovyscript.compat.mods.thaumcraft.DustTrigger
+import com.cleanroommc.groovyscript.compat.mods.thaumcraft.SmeltingBonus
+import com.cleanroommc.groovyscript.compat.mods.thaumcraft.Research
+
+mods.thaumcraft.Crucible.removeByOutput(item('minecraft:gunpowder'))
+
+mods.thaumcraft.Crucible.recipeBuilder()
+    .catalyst(item('minecraft:rotten_flesh'))
+    .output(item('minecraft:gold_ingot'))
+    .researchKey("UNLOCKALCHEMY@3")
+    .aspects((new AspectList()).add(Aspect.METAL, 20))
+    .register()
+
+mods.thaumcraft.InfusionCrafting.removeByOutput(item('thaumcraft:crystal_terra'))
+
+mods.thaumcraft.InfusionCrafting.recipeBuilder()
+    .input(item('minecraft:gunpowder'))
+    .output(item('minecraft:gold_ingot'))
+    .researchKey("UNLOCKALCHEMY@3")
+    .aspects((new AspectList()).add(Aspect.METAL, 20))
+    .instability(10)
+    .components(new Object[]{ThaumcraftApiHelper.makeCrystal(Aspect.AIR), ThaumcraftApiHelper.makeCrystal(Aspect.FIRE), ThaumcraftApiHelper.makeCrystal(Aspect.WATER), ThaumcraftApiHelper.makeCrystal(Aspect.EARTH), ThaumcraftApiHelper.makeCrystal(Aspect.ORDER)})
```

Comment:
Pls use `List<Object>` instead of arrays.
Might be worth to make a `crystal()` bracket handler or something like `mods.thaumcraft.air_crystal` etc.

Reply:
crystal() bracket handler added in commit:  [https://github.com/CleanroomMC/GroovyScript/commit/bbb0808794642b9db2a112ac48477a96781bb90f](https://github.com/CleanroomMC/GroovyScript/commit/bbb0808794642b9db2a112ac48477a96781bb90f)